cmd/checkpoint/main.go

```diff
@@ -47,49 +48,73 @@ func main() {
 		glog.Fatalf("failed to load existing checkpoint manifests: %v", err)
 	}
 	glog.Info("begin pods checkpointing...")
-	run(kubeAPIServer, tempAPIServer, api.NamespaceSystem, checkpoints)
+	run(api.NamespaceSystem, checkpoints)
 }
 
-func run(actualPodName, tempPodName, namespace string, checkpoints map[string]struct{}) {
+func run(namespace string, checkpoints map[string]struct{}) {
 	client := newAPIClient()
 	for {
-		_, checkpointed := checkpoints[checkpointManifest(tempPodName)]
-
 		var podList v1.PodList
 		if err := json.Unmarshal(getPodsFromKubeletAPI(), &podList); err != nil {
 			glog.Fatal(err)
 		}
-		switch {
-		case bothRunning(podList, actualPodName, tempPodName, namespace):
-			glog.Infof("both temp %v and actual %v pods running, removing temp pod", actualPodName, tempPodName)
-			// Both the temp and actual pods are running.
-			// Remove the temp manifest from the config dir so that the
-			// kubelet will stop it.
-			if err := os.Remove(activeManifest(tempPodName)); err != nil {
-				glog.Error(err)
+
+		pwc := getPodsWithCheckpointAnnotation(podList)
+		// special case for handling API server readiness
+		if _, ok := pwc[kubeAPIServer]; ok {
+			// Make sure it's actually running. Sometimes we get that
+			// pod manifest back, but the server is not actually running.
+			if _, err := client.Discovery().ServerVersion(); err != nil {
```

Comment:
This could be incorrect when running multiple api-servers. The local pod we're trying to checkpoint could in fact be broken, but we happen to hit a different api-server using this client request.

This logic was already in place, so we don't need to fix it here. But maybe can you add a "TODO: use pod healthcheck to determine if apiserver pod is successfully running."